src/DynamoCore/Engine/CodeGeneration/AstBuilder.cs

```diff
@@ -1,125 +1,36 @@
 {
-    /// <summary>
-    /// Returns notification for AST compilation events.                                                 
-    /// </summary>
-    public interface IAstNodeContainer
-    {
-        /// <summary>
-        /// Indicates to start compiling a NodeModel to AST nodes.
-        /// </summary>
-        /// <param name="nodeGuid"></param>
-        void OnCompiling(Guid nodeGuid);
-
-        /// <summary>
-        /// Indicates a NodeModel has been compiled to AST nodes. 
-        /// </summary>
-        /// <param name="nodeGuid"></param>
-        /// <param name="astNodes"></param>
-        void OnCompiled(Guid nodeGuid, IEnumerable<AssociativeNode> astNodes);
-    }
-
-    /// <summary>
-    /// This event is triggerred when compiling a NodeModel to AST nodes.
-    /// </summary>
-    public class CompilingEventArgs : EventArgs
-    {
-        /// Construct ASTBuildingEventArgs with NodeModel.
-        public CompilingEventArgs(Guid node)
-        {
-            Node = node;
-        }
-
-        /// <summary>
-        /// Guid of NodeModel that is being compiled to AST.
-        /// </summary>
-        public Guid Node { get; private set; }
-    }
-
-    /// <summary>
-    /// This event is triggered when a NodeModel has been compiled to a list of
-    /// AST nodes.
-    /// </summary>
-    public class CompiledEventArgs : EventArgs
-    {
-        /// <summary>
-        /// Construct ASTBuiltEventArgs with NodeModel and AST nodes.
-        /// </summary>
-        /// <param name="node"></param>
-        /// <param name="astNodes"></param>
-        internal CompiledEventArgs(Guid node, IEnumerable<AssociativeNode> astNodes)
-        {
-            Node = node;
-            AstNodes = astNodes;
-        }
-
-        /// <summary>
-        /// Guid of node that has been built to AST nodes.
-        /// </summary>
-        public Guid Node { get; private set; }
-
-        /// <summary>
-        /// Built AST nodes.
-        /// </summary>
-        public IEnumerable<AssociativeNode> AstNodes { get; private set; }
-    }
-
-    /// <summary>
-    /// The context of AST compilation
-    /// </summary>
-    public enum CompilationContext
-    {
-        /// <summary>
-        /// No specific context.
-        /// </summary>
-        None,
-
-        /// <summary>
-        /// Compiled AST nodes finally will be executed.
-        /// </summary>
-        DeltaExecution,
-
-        /// <summary>
-        /// Compiled AST nodes used in node to code.
-        /// </summary>
-        NodeToCode,
-
-        /// <summary>
-        /// Compiled AST nodes used in previwing graph.
-        /// </summary>
-        PreviewGraph,
-    }
-
     /// <summary>
     /// AstBuilder is a factory class to create different kinds of AST nodes.
     /// </summary>
     public class AstBuilder : LogSourceBase
     {
         private readonly IAstNodeContainer nodeContainer;
 
+        internal ProfilingSession profilingSession = null;
```

Comment:
is this internal for a specific reason? (sorry have not looked further down to see if that's the case yet)

Reply:
No, has been changed to be private now